### PR TITLE
Make table time_format configurable

### PR DESCRIPTION
**Pull request by patrixr**
## Issue

When using the plugin with MySQL, the date comparison gets rejected due to the timezone.

Screenshot 2021-09-30 at 2 40 10 PM

Running the following SQL with a timezone for `last_update_column_value` would cause this

```sql
SELECT * FROM table WHERE update_column > last_update_column_value
```

## Fix

This PR makes the time format configurable, allowing us to override it **per table.**

Example configuration

```xml
<source>
  @type sql
   
   # ...
  
  <table>
    table my_table
    tag my_tag
    update_column updated_at
    time_format %Y-%m-%d %H:%M:%S.%6N
  </table>
</source>
```

Note: I kept the pre-defined format as it was, to avoid any backwards compatibility issue

@repeatedly

### CODE REVIEW

File: lib/fluent/plugin/in_sql.rb
Comment (cosmo0920) [inline lib/fluent/plugin/in_sql.rb:80]: We wouldn't use class variable.
For reloading feature support, please use variable store instead:
https://github.com/fluent/fluentd/blob/master/lib/fluent/variable_store.rb

**Review by cosmo0920 — CHANGES_REQUESTED** [review]

**Comment by cosmo0920** [comment]
And follow DCO. 
Please add Signed-off-by line in your commit.
See: https://github.com/fluent/fluent-plugin-sql/pull/112/checks?check_run_id=3785622978

**Comment by patrixr** [comment]
@cosmo0920 I've updated the PR with the VariableStore, let me know if that's okay. (Also signed off the commits and documented the option in the readme)

File: VERSION
Comment (cosmo0920) [inline VERSION:1]: Could you revert this line?
Gem version shouldn't be touched in PR.

**Review by cosmo0920 — CHANGES_REQUESTED** [review]

**Comment by cosmo0920** [comment]
Thanks for your work. But gem version should not bump in this PR.
Bumping version is maintainer matter.

**Comment by cosmo0920** [comment]
@ashie @kenhys Could you take a look, too? This PR is almost reasonable for me except fort bumping version line.

**Comment by kenhys** [comment]
Looks nice.

**Comment by patrixr** [comment]
@cosmo0920 Sure thing, I'll revert the version number : )

**Comment by kenhys** [comment]
@patrixr could you squash needless commit?

**Comment by patrixr** [comment]
@kenhys done. Squashed into a single commit

File: lib/fluent/plugin/in_sql.rb
Comment (ashie) [inline lib/fluent/plugin/in_sql.rb]: I'm not comfortable using `Fluent::VariableStore`.
Because it's like a global variable, we should treat it carefully.
For example, the name `:time_formats` isn't appropriate, it might be used by other modules.
In Fluentd code, plugin name is used as name space for `Fluent::VariableStore`. e.g.) `:in_tail`
We should follow such naming convention.

On the other hand, I think we don't need to using `Fluent::VariableStore` here, setting a constant to the `@model` class is enough. I've added a commit for it: 9dee029f716d904ec14ae7797f17233b2ac76a80
Please check it @patrixr @cosmo0920 
I'll approve this pull request if it's no problem for you.

Comment (patrixr) [reply lib/fluent/plugin/in_sql.rb]: Agreed @ashie, I much prefer this version 👍

**Review by ashie — APPROVED** [review]

**Pull request merged**
